### DISCUSSION TITLE
[SOLVED] What is the proper way to add Shield to my App

### DISCUSSION THREAD

**Discussion by foxbille** [codeigniter4/shield#441] [Q&A] [answered]
Hello,
Have installed CI4 AppStarter, works well, added Shield, works well (Setup done, successful access to specific shield pages : Login, register and so on)
I wanted to restrict access to page Home provided with CI4 installation
Added  :
```php
['session' => ['except' => ['login*', 'register*']]],
```
to `$filters in \App\Config\filters.php`
That's all ! 

doesn't send the login page, provides home page
Want did I forgot to add to link and use Shield protection services ?
I did not found any small tuto about a simple use of services provided by Shield, but lots of help about how to manage Shield.Which i don't care for now.
Does any one know where I could find a bit of starting help ?
Thanks a lot !
Eric

**Comment by datamweb** [comment, marked as answer]
> doesn't send the login page, provides home page
> Want did I forgot to add to link and use Shield protection services ?
That's all you have to do.

`$globals in \App\Config\filters.php`
```php
    /**
     * List of filter aliases that are always
     * applied before and after every request.
     *
     * @var array
     */
    public $globals = [
        'before' => [
            // 'honeypot',
            // 'csrf',
            // 'invalidchars',
              'session' => ['except' => ['login*', 'register', 'auth/a/*']],
        ],
        'after' => [
            'toolbar',
            // 'honeypot',
            // 'secureheaders',
        ],
    ];
```
Now if you see `http://localhost:8080/` you should be redirected to page `http://localhost:8080/index.php/login`.

What do you mean, can you explain clearly?

**Reply by kenjis** [reply]
@datamweb The sample config in the docs is not correct:
https://github.com/codeigniter4/shield/blob/develop/docs/install.md#protect-all-pages

`$filter` does not have `except`:
https://codeigniter4.github.io/CodeIgniter4/incoming/filters.html#filters
so it is better to use `$globals`:
https://codeigniter4.github.io/CodeIgniter4/incoming/filters.html#globals

**Reply by datamweb** [reply]
@kenjis Thanks, I sent #443 .

**Reply by foxbille** [reply]
That's exactly what I wanted : a way to use authentication in my web application.
Now it works : My request is re-directed to login.
Thanks a lot for your help
Eric
